libraries/AP_HAL_ChibiOS/RCInput.cpp

```diff
@@ -75,12 +75,11 @@ bool RCInput::new_input()
     if (!_init) {
         return false;
     }
+
     bool valid;
-    {
-        WITH_SEMAPHORE(rcin_mutex);
-        valid = _rcin_timestamp_last_signal != _last_read;
-        _last_read = _rcin_timestamp_last_signal;
-    }
```

Comment:
The brackets constrain the scope of the semaphore, I don't think this change is correct.